autobot/github.py

```diff
@@ -17,232 +17,217 @@
+        reviews = [r for r in self.reviews()]
+        return len(reviews) == 0
+
+    filters = [is_open, can_close, needs_comment, can_merge, needs_review]
+
+    @property
+    def status(self):
+        """The status of the pull request."""
+        status = {}
+        for f in self.filters:
+            status[f.__name__] = f(self)
+        return status
+
+    @property
+    def actions(self):
+        """The suitable actions for the pull request."""
+        actions = []
+        if not self.status["is_open"]:
+            return actions
+        if self.status["needs_review"]:
+            actions.append("Review this!")
+        elif self.status["can_merge"]:
+            actions.append("Merge this!")
+        elif self.status["can_close"]:
+            actions.append("Close this!")
+        elif self.statusp["needs_comment"]:
+            actions.append("Comment on this!")
+        return actions
+
+
+class Issue(GHWrapper):
+    """Issue wrapper object."""
+
+    def __init__(self, issue, maintainers=None):
+        """Issue wrapper object initialization."""
+        self._gh_obj = issue
+        self.maintainers = maintainers
+
+    @property
+    def info(self):
         """Fetch information for an issue."""
         return {
-            "id": issue.number,
-            "url": issue.html_url,
-            "title": issue.title,
-            "creation_date": issue.created_at,
-            "description": issue.body,
-            "state": issue.state,
+            "id": self.number,
+            "url": self.html_url,
+            "title": self.title,
+            "creation_date": self.created_at,
+            "description": self.body,
+            "state": self.state,
+            "user": {"name": self.user.login, "url": self.user.html_url},
             "labels": [
                 {"name": label.name, "color": label.color, "url": label.url}
-                for label in issue.labels()
+                for label in self.labels()
             ],
-            "user": {"name": issue.user.login, "url": issue.user.url},
         }
 
-    def fetch_repo_info(self, repo):
+    def is_open(self):
+        """Check if issue is open."""
+        return self.state == "open"
+
+    def can_close(self):
+        """Check if issue can be closed."""
+        return (
+            datetime.utcnow().replace(tzinfo=pytz.utc) - self.updated_at
+        ).days >= 3 * 30
+
+    def needs_comment(self):
+        """Check if issue needs comment."""
+        comments = [comment for comment in self.comments()]
+        return not comments or comments[-1].user.login not in self.maintainers
+
+    def lbls(self):
+        """Fetch issue's labels."""
+        return [l.name for l in self.labels()]
+
+    filters = [is_open, can_close, needs_comment, lbls]
+
+    @property
+    def status(self):
+        """The status of the issue."""
+        status = {}
+        for f in self.filters:
+            status[f.__name__] = f(self)
```

Comment:
Instead of that, `filters` could be a dictionary with names or be only present inside this function, since it's only used here:

```python
filters = {
    'is_open': self.is_open,
    'can_close': self.can_close,
    'needs_comment': self.needs_comment,
    ...
}
```